docs/data/date-pickers/custom-field/PickerWithBrowserField.tsx

```diff
@@ -82,46 +78,18 @@ interface BrowserDateFieldProps
 
 const BrowserDateField = React.forwardRef(
   (props: BrowserDateFieldProps, ref: React.Ref<HTMLDivElement>) => {
-    const {
-      inputRef: externalInputRef,
+    const { slots, slotProps, ...textFieldProps } = props;
+
+    const fieldResponse = useDateField<Dayjs, typeof textFieldProps>(textFieldProps);
+
+    /* If you don't need a clear button, you can skip the use of this hook */
+    const processedFieldProps = useClearableField({
+      props: fieldResponse,
       slots,
       slotProps,
```

Comment:
What's the reason for doing
```js
useClearableField({ props: fieldResponse, slots, slotProps, })
```

instead of 

```js
useClearableField({ fieldResponse, slots: { ...fieldResponse.slots, ...slots }, slotProps: { ...fieldResponse.slotProps, ...slotProps } })
```

In this case with `slots` it's eavier, but it general, it looks weird to have `props` and `slots` knowing, that `slots` is also a prop 🤔

Reply:
I tried to do it and it was making our internal implementation more complex because we are passing almost all the props to `useSlotProps` and I don't think we can to pass the slots themselves.

So we would have to do something like:

```tsx
```